pkg/kubelet/kubelet_node_status.go

```diff
@@ -579,7 +579,12 @@ func (kl *Kubelet) tryUpdateNodeStatus(ctx context.Context, tryNumber int) error
 	}
 
 	node, changed := kl.updateNode(ctx, originalNode)
-	shouldPatchNodeStatus := changed || kl.clock.Since(kl.lastStatusReportTime) >= kl.nodeStatusReportFrequency
+	shouldPatchNodeStatus := changed || kl.clock.Since(kl.lastStatusReportTime) >= kl.nodeStatusReportFrequency+kl.delayAfter1stNodeStatusUpdate
+
+	// Only reset nodeStatusReportWaitDurationAfterInitialUpdate to 0 when it is used once.
+	if !kl.lastStatusReportTime.IsZero() && shouldPatchNodeStatus && !changed {
+		kl.delayAfter1stNodeStatusUpdate = 0
```

Comment:
with this logic we are not fixing a situation when many nodes come back to ready at the same time. I thought we want to add randomness after every time `changed` was `true`